app/models/piece.rb

```diff
@@ -1,3 +1,82 @@
 class Piece < ApplicationRecord
-	belongs_to :game
+  belongs_to :game
+
+  def is_obstructed?(new_x, new_y)
+    x_converter = { 'a' => 1, 'b' => 2, 'c' => 3, 'd' => 4, 'e' => 5, 'f' => 6, 'g' => 7, 'h' => 8 }
+    x_searcher = { 1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd', 5 => 'e', 6 => 'f', 7 => 'g', 8 => 'h' }
+    new_x = x_converter[new_x]
+    new_y = new_y.to_i
+    old_x = x_converter[self.x_coordinate]
+    old_y = y_coordinate.to_i
+    line_type = self.direction_check(new_x, new_y, old_x, old_y)
+    if line_type == 'invalid'
+
+      return true
+
+    elsif line_type == 'horizontal'
```

Comment:
@glassrenegade kudos on the implementation! A few ideas how to refactor the code and make it a bit simpler. we could abstract the elseif statements to separate methods so that the overall structure would be sth like this:

def is_obsctucted?(new_x, new_y)
  # your setup
  # some conditionals
   is_horizontally_obstructed?(new_x, new_y) 
  # some conditionals
   is_vertically_obstructed?(new_x, new_y) 
#... etc
end


Once this is done we can do even more - let's talk about it during the session :)